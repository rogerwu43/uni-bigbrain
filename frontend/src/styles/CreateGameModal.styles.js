import styled from 'styled-components';

export const CreateGameModal = styled.div`
  background: white;
  display: ${props => props.active ? 'block' : 'none'};
  padding: 10px 10px 30px;
  position: fixed;
  height: auto;
  left:50%;
  max-width: 900px;
  top:50%;
  transform: translate(-50%,-50%);
  width: 90%;
`;

export const ModalBackground = styled.div`
  background: rgba(0, 0, 0, 0.6);
  display: ${props => props.active ? 'block' : 'none'};position: fixed;
  height: 100%;
  left: 0;
  top: 0;
  width:100%;
`;

export const FormError = styled.p`
  color: red;
`;
