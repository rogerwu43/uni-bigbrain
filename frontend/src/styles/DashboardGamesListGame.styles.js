import styled from 'styled-components';

export const DashboardGamesListGame = styled.div`
  background-color: white;
  display: flex;
  flex-direction: column;
  height: 150px;
  margin: 10px 0px 10px;
`;

export const Data = styled.div`
  display: flex;
  flex: 1;
  flex-direction: column;
  overflow: hidden;
`;

export const Details = styled.div`
  display: flex;
  height: 100px;
  padding: 5px;
`;

export const Options = styled.div`
  align-items: center;
  display: flex;
  height: 50px;
  justify-content: flex-start;
  padding: 0px 5px 5px;
`;

export const SubDetails = styled.div`
  align-items: center;
  background-color: #F2F2F2;
  display: flex;
  height: 50px;
  justify-content: flex-start;

  & > p {
    background-color: #6E6E6E;
    border-radius: 14px;
    color: white;
    font-size: 12px;
    margin: 0 15px;
    padding: 1px 10px 2px;
  }
`;

export const Thumbnail = styled.div`
  align-items: flex-end;
  background-color: #CCCCCC;
  background-image: url(${props => props.img});
  background-position: center;
  background-repeat: no-repeat;
  background-size: 100px;
  display: flex;
  height: 100px;
  justify-content: center;
  margin: 0px 5px 0px 0px;
  min-height: 100px;
  min-width: 100px;
  width: 100px;

  & > p {
    background-color: #000000;
    color: white;
    font-size: 14px;
    margin: 0px;
    padding: 5px 0px;
    width: 100%;
  }
`;

export const DefaultThumbnail = styled(Thumbnail)`
  background-size: 50px;
`;

export const Title = styled.div`
  align-items: center;
  display: flex;
  height: 50px;
  margin: 0px 0px 5px 0px;

  & > p {
    font-size: 20px;
    font-weight: 600;
    margin: 0px;
    overflow: hidden;
    padding: 0px 20px 0px 10px;
    text-overflow: ellipsis;
    white-space: nowrap;
  }
`;
