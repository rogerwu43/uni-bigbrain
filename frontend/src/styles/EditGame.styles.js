import styled from 'styled-components';

export const EditGameQuestion = styled.div`
  align-items: center;
  background-color: #FFFFFF;
  display: flex;
  height: 44px;
  margin: 5px 0px;
  overflow: hidden;

  & > div {
    padding: 0px 6px;
  }

  & > p {
    flex: 1;
    font-weight: 600;
    overflow: hidden;
    padding-left: 15px;
    text-align: left;
    text-overflow: ellipsis;
    white-space: nowrap;
  }
`;

export const EditName = styled.i`
  font-size: 20px;
  padding: 5px;

  &:hover {
    background-color: #C1C1C1;
    border-radius: 10px;
    cursor: pointer;
    opacity: 0.6;
  }
`;

export const GameThumbnail = styled.img`
  display: block;
  margin: auto;
  margin-bottom: 10px;
  max-height: 50px;
  max-width: 50px;
`;

export const ThumbnailForm = styled.form`
  margin: 20px 10px;
  text-align: left;

  & > button {
    display: block;
  }

  & > img {
    margin-bottom: 10px;
    max-height: 100px;
    max-width: 100px;
  }

  & > input {
    display: block;
    margin-bottom: 15px;
  }

  & > p {
    margin: 0px 0px 10px;
  }
`;

export const TitleForm = styled.div`
  align-items: center;
  display: flex;
  flex-direction: column;
  margin: auto;
  width: 100%;
`;

export const TitleInput = styled.input`
  border: 1px solid lightgrey;
  border-radius: 5px;
  font-family: Quicksand, sans-serif;
  font-size: 14px;
  font-weight: 400;
  padding: 10px;
  text-align: center;
  width: 100%;
`;
