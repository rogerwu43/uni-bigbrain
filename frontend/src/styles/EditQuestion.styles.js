import styled from 'styled-components';

// Divs
export const AnswerDiv = styled.div`
  align-items: center;
  display: flex;
  justify-content: center;
  width: 100%;
;`

export const AnswerOptionList = styled.div`
  align-items: center;
  display: flex;
  flex-direction: column;
  justify-content: center;
  width: 80%;
`;

export const CurrentImage = styled.div`
  align-items: center;
  border: 1px dashed grey;
  display: flex;
  flex-direction: column;
  margin-bottom: 10px;
  padding: 10px;

  button + button {
    margin-top: 10px;
  }

  & > img {
    max-height: 200px;
    max-width: 200px;
  }

  & > input {
    font-family: Quicksand, sans-serif;
    margin-bottom: 10px;
  }
`;

export const CurrentVideo = styled.div`
  align-items: center;
  border: 1px dashed grey;
  display: flex;
  flex-direction: column;
  margin-bottom: 10px;
  padding: 10px;

  button + button {
    margin-top: 10px;
  }

  & > iframe {
    margin-bottom: 10px;
    max-height: 200px;
    max-width: 200px;
  }

  & > img {
    margin-bottom: 10px;
    max-height: 200px;
    max-width: 200px;
  }

  & > input {
    font-family: Quicksand, sans-serif;
    height: 24px;
    margin-bottom: 10px;
    width: 80%;
  }
`;

export const HorizontalDiv = styled.div`
  border-bottom: 1px solid lightgrey;
  margin: 10px auto;
  width: 80%;
`;

export const InputOptions = styled.div`
  align-items: center;
  display: flex;
  flex-direction: column;
  margin-bottom: 20px;
`;

export const MediaOptions = styled.div`
  display: flex;
  flex-direction: column;
  justify-content: center;
  margin-bottom: 20px;
  width: 80%;
`;

export const StatOptions = styled.div`
  align-items: center;
  display: flex;
  flex-direction: column;

  & > p {
    font-size: 14px;
    font-weight: 600;
    margin: 0px 0px 5px 0px;
  }

  & > select {
    border: none;
    font-family: Quicksand, sans-serif;
    font-size: 12px;
    padding: 10px;
    width: 200px;
  }

  p + select {
    margin-bottom: 20px;
  }
`;

// buttons
export const OptionsButton = styled.button`
  border: none;
  border-radius: 5px;
  color: white;
  display: inline-block;
  font-family: 'Quicksand', sans-serif;
  font-size: 14px;
  font-weight: 600;
  padding: 5px 15px 6px 12px;
  text-align: center;
  text-decoration: none;

  &:hover {
    cursor: pointer;
  }

  & > i {
    margin-right: 10px;
  }

  & + & {
    margin-bottom: 10px;
  }
`;

export const DiscardButton = styled(OptionsButton)`
  background-color: #D72619;
  font-size: 12px;

  &:hover {
    background-color: #E93C30;
  }
`;

export const SaveButton = styled(OptionsButton)`
  background-color: #4CAF50;
  font-size: 24px;
  margin-bottom: 10px;
  padding: 20px 50px;

  &:hover {
    background-color: #5DC661;
  }
`;

// Forms
export const EditQuestion = styled.form`
  margin: auto;
  max-width: 960px;
  padding: 10px 0px;

  @media (max-width: 980px) {
    margin: 0px 10px;
  }
`;

// inputs
export const AnswerInput = styled.input`
  border: 1px solid lightgrey;
  border-radius: 5px;
  font-family: Quicksand, sans-serif;
  font-size: 14px;
  font-weight: 400;
  margin-bottom: 10px;
  padding: 10px;
  text-align: center;
  width: 80%;
`;

export const QuestionInput = styled.input`
  border: 1px solid lightgrey;
  border-radius: 5px;
  font-family: Quicksand, sans-serif;
  font-size: 20px;
  font-weight: 600;
  margin-bottom: 20px;
  padding: 10px;
  text-align: center;
  width: 80%;
`;

// p
export const FormError = styled.p`
  color: red;
  font-style: italic;
  text-align: center;
`;

export const Select = styled.select`
  background-color: white;
`;
