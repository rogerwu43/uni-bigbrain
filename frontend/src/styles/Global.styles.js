import { Link } from 'react-router-dom';
import styled from 'styled-components';

import lobbyBackground from '../images/lobby-background.png';

// Parents (Level 0).

export const ErrorText = styled.p`
  color: red;
  font-size: 14px;
  font-style: italic;
  font-weight: 500px;
`;

export const HorizontalDiv = styled.div`
  border-bottom: 1px solid lightgrey;
  margin: 10px auto;
  width: 100%;
`;

export const LargeCreateButton = styled.button`
  background-color: #0A43B5;
  border: none;
  border-radius: 30px;
  color: white;
  display: inline-block;
  font-family: 'Quicksand', sans-serif;
  font-size: 20px;
  font-weight: 700;
  margin: 20px 0;
  padding: 10px 21px 11px 19px;
  text-align: center;
  text-decoration: none;

  &:focus {
    outline: none;
  }

  &:hover {
    background-color: #285ECA;
    cursor: pointer;
  }

  & > i {
    margin-right: 10px;
  }
`;

export const MediumButton = styled.button`
  border: none;
  border-radius: 5px;
  color: #7A7A7A;
  display: inline-block;
  font-family: 'Quicksand', sans-serif;
  font-size: 11px;
  font-weight: 600;
  height: 30px;
  margin: 0 10px;
  padding: 5px 12px 6px;
  text-align: center;
  text-decoration: none;

  &:focus {
    outline: none;
  }

  &:hover {
    cursor: pointer;
  }

  & > i {
    font-size: 12px;
    margin-right: 10px;
  }
`;

export const MiniLink = styled(Link)`
  color: blue;
  font-size: 14px;

  &:visited {
    color: blue;
  }
`;

export const PageTitle = styled.h1`
  border-color: grey;
  border-style: solid;
  border-width: 1px 0px;
  display: inline-block;
  font-weight: 700;
  padding: 10px 30px;
`;

export const PageWrapper = styled.div`
  background-image: url(${props => props.background ? lobbyBackground : ''});
  height: calc(100% - 50px);
  margin: auto;
  max-width: 960px;
  text-align: center;

  @media (max-width: 980px) {
    margin: 0 10px;
  }
`;

export const SectionTitle = styled.h2`
  font-weight: 700;
  margin: 20px 0px;
`;

export const SubSectionTitle = styled.h3`
  font-weight: 600;
  margin: 20px 0;
`;

export const SubText = styled.p`
  font-size: 12px;
  font-weight: 400;
`;

// Inheritors (Level 1).

export const BlueMediumButton = styled(MediumButton)`
  background-color: #3D7DD0;
  color: white;
  margin: 0 5px;

  &:hover {
    background-color: #3E8FF3;
  }
`;

export const DarkGreenMediumButton = styled(MediumButton)`
  background-color: #5DAA80;
  color: white;
  margin: 0 5px;

  &:hover {
    cursor: default;
  }
`;

export const GreenMediumButton = styled(MediumButton)`
  background-color: #4CAF50;
  color: white;
  margin: 0 5px;

  &:hover {
    background-color: #5DC661;
  }
`;

export const OrangeMediumButton = styled(MediumButton)`
  background-color: #EB3115;
  color: white;
  margin: 0 5px;

  &:hover {
    background-color: #FF3B1C;
  }
`;

export const RedMediumButton = styled(MediumButton)`
  background-color: #D72619;
  color: white;
  margin: 0 5px;

  &:hover {
    background-color: #E93C30;
  }
`;
