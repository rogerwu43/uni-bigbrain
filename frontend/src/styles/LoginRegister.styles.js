import styled from 'styled-components';

export const LabelInput = styled.div`
  margin-bottom: 15px;
  text-align: left;

  & > label {
    font-size: 14px;
    font-weight: 500;
  }
`;

export const LoginDiv = styled.div`
  align-items: center;
  display: flex;
  flex-direction: column;
`;

export const LoginForm = styled.form`
  align-items: center;
  display: flex;
  flex-direction: column;
`;
