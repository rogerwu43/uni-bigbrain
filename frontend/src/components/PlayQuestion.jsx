import PropTypes from 'prop-types';
import React from 'react';

import * as GS from '../styles/Global.styles';
import * as S from '../styles/Play.styles';
import api from '../util/Api';

function PlayQuestion ({ questionData, answersData, showPopup, pid }) {
  const answers = questionData.question.answers;
  const [selected, setSelected] = React.useState([]);
  const [timer, setTimer] = React.useState(0);
  const [questionEnds, setQuestionEnds] = React.useState(-1);

  async function updateSelected (id) {
    if (questionData.question.type === 'Single') {
      setSelected([id]);
      const response = await api.putPlayPlayeridAnswer(pid, [id])
      if (response.status !== 200) {
        const error = await response.json();
        showPopup('Error: ' + error.error);
      }
    } else {
      const newSelected = [...selected];
      if (newSelected.includes(id)) {
        const index = newSelected.indexOf(id);
        if (index > -1) {
          newSelected.splice(index, 1);
        }
      } else {
        newSelected.push(id);
      }
      setSelected(newSelected)
      const response = await api.putPlayPlayeridAnswer(pid, newSelected)
      if (response.status !== 200 && newSelected.length !== 0) {
        const error = await response.json();
        showPopup('Error: ' + error.error);
      }
    }
  }

  const children = answers.map((val) => {
    if (answersData.length === 0) {
      return (
        <S.AnswerOption key={val.id}>
          <p>{val.answer}</p>
          {selected.includes(val.id)
            ? <S.SelectedAnswer onClick={(e) => { updateSelected(val.id); }}>Selected!</S.SelectedAnswer>
            : <S.NotSelectedAnswer onClick={(e) => { updateSelected(val.id); }}>Not Selected</S.NotSelectedAnswer>
          }
        </S.AnswerOption>
      );
    } else {
      if (answersData.includes(val.id)) {
        return (
          <div key={val.id}>
            <p><b>{val.answer}</b> was correct.</p>
          </div>
        )
      } else {
        return (
          <div key={val.id}>
            <p><b>{val.answer}</b> was not correct.</p>
          </div>
        )
      }
    }
  });

  // Set remaining question time on page render.
  React.useEffect(() => {
    const questionStarted = Date.parse(questionData.question.isoTimeLastQuestionStarted);
    const questionEnds = questionStarted + (questionData.question.timeLimit) * 1000;
    setQuestionEnds(questionEnds);
    const timeLeft = Math.floor((questionEnds - Date.now()) / 1000);
    if (timeLeft > 0) {
      setTimer(timeLeft);
    }
  }, [questionData]);

  React.useEffect(() => {
    setSelected([])
  }, [questionData.question.id, answersData.length]);

  React.useEffect(() => {
    const timerInterval = timer > 0 && setInterval(() => {
      setTimer(Math.floor((questionEnds - Date.now()) / 1000));
    }, 1000);
    return () => clearInterval(timerInterval);
  }, [timer]);

  return (
    <div>
      <S.Question>{questionData.question.question}</S.Question>
      <GS.HorizontalDiv/>
      <S.Media>
        {questionData.question.image !== ''
          ? <S.Image src={questionData.question.image}/>
          : null
        }
        {questionData.question.video !== ''
          ? <S.Video src={questionData.question.video}/>
          : null
        }
      </S.Media>
      {questionData.question.type === 'Single'
        ? <GS.SubSectionTitle>Select 1 Answer</GS.SubSectionTitle>
        : <GS.SubSectionTitle>Select Up To {questionData.question.answers.length} Answers</GS.SubSectionTitle>
      }
      <GS.HorizontalDiv/>
      {timer === 0
        ? <S.Timer>Time Out!</S.Timer>
        : <S.Timer>{timer}</S.Timer>
      }
      {children}
    </div>
  );
}

export default PlayQuestion;

PlayQuestion.propTypes = {
  pid: PropTypes.number,
  questionData: PropTypes.object,
  answersData: PropTypes.array,
  showPopup: PropTypes.func
}
