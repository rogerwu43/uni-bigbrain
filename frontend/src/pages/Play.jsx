import React from 'react';
import { Redirect } from 'react-router-dom';
import PropTypes from 'prop-types';

import PlayQuestion from '../components/PlayQuestion';
import PlayLobby from '../components/PlayLobby';
import PlayResults from '../components/PlayResults';
import * as GS from '../styles/Global.styles';
import api from '../util/Api';
import history from '../util/History';

function PlayPage ({ showPopup }) {
  const [questionData, setQuestionData] = React.useState('');
  const [answersData, setAnswersData] = React.useState([]);
  const [resultsData, setResultsData] = React.useState('');
  const [body, setBody] = React.useState('lobby');
  const [questionPoints] = React.useState({});
  const [timeLimits] = React.useState({});

  const pid = history.location.state?.pid
  if (pid === undefined) {
    return <Redirect to='/join'/>
  }
  // poll
  async function getData () {
    const statusResponse = await api.getPlayPlayeridStatus(pid);
    if (statusResponse.status === 200) {
      const status = await statusResponse.json();
      // show question
      if (status.started === true) {
        const answersResponse = await api.getPlayPlayeridAnswer(pid);
        if (answersResponse.status === 200) {
          const answersData = await answersResponse.json();
          setAnswersData(answersData.answerIds);
        } else {
          setAnswersData([]);
        }
        const questionResponse = await api.getPlayPlayeridQuestion(pid);
        const questionData = await questionResponse.json();
        setQuestionData(questionData);
        questionPoints[questionData.question.id] = questionData.question.points;
        timeLimits[questionData.question.id] = questionData.question.timeLimit;
        setBody('question');
      // show lobby
      } else {
        setBody('lobby');
      }
      // change i think
      if (window.location.pathname === '/play') { setTimeout(getData, 2000); }
    } else {
      const error = await statusResponse.json();
      // show results if "Session ID is not an active session"
      const resultsResponse = await api.getPlayPlayeridResults(pid);
      if (resultsResponse.status === 200) {
        const resultsData = await resultsResponse.json();
        setResultsData(resultsData);
        setBody('results');
      } else {
        // actual error
        showPopup('Error: ' + error.error)
      }
    }
  }

  // stop polling on back
  window.onpopstate = () => {
    history.go();
  };

  React.useEffect(() => {
    getData();
  }, []);

  return (
    <>
      {body === 'lobby'
        ? <GS.PageWrapper background>
            <PlayLobby/>
          </GS.PageWrapper>
        : <GS.PageWrapper>
            {body === 'question'
              ? <PlayQuestion questionData = {questionData} answersData = {answersData} pid = {pid} showPopup={showPopup} />
              : null
            }
            {body === 'results'
              ? <PlayResults resultsData = {resultsData} pid = {pid} questionPoints = {questionPoints} timeLimits = {timeLimits} />
              : null
            }
          </GS.PageWrapper>
      }
    </>
  );
}

PlayPage.propTypes = {
  showPopup: PropTypes.func
}

export default PlayPage;
