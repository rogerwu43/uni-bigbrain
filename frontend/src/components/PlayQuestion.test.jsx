import { mount } from 'enzyme'
import React from 'React'
import renderer from 'react-test-renderer'
import PlayQuestion from './PlayQuestion'
import * as S from '../styles/Play.styles';

describe('PlayQuestion', () => {
  const noop = () => {};

  const questionData = {
    question: {
      question: 'Cats or dogs',
      id: 1,
      type: 'Single',
      answers: [
        { answer: 'cats', id: 1, isCorrect: true },
        { answer: 'dogs', id: 2, isCorrect: false },
      ],
      timeLimit: '60',
      points: 1,
      image: '',
      video: ''
    }
  }
  const emptyAnswersData = []
  const answersData = [1]

  it('displays the question passed in with the prop', () => {
    const question = mount(<PlayQuestion questionData={questionData} answersData={emptyAnswersData} showPopup={noop} />);
    expect(question.find(S.Question)).toHaveLength(1);
    expect(question.find(S.Question).text()).toBe('Cats or dogs');
    const answers = question.find(S.AnswerOption);
    expect(answers).toHaveLength(2);
    expect(answers.at(0).find('p').text()).toBe('cats');
    expect(answers.at(1).find('p').text()).toBe('dogs');
  });

  it('changes the display of the button when selected', () => {
    const question = mount(<PlayQuestion questionData={questionData} answersData={emptyAnswersData} showPopup={noop} />);
    expect(question.find(S.AnswerOption).at(0).find(S.NotSelectedAnswer)).toHaveLength(1);
    expect(question.find(S.AnswerOption).at(0).find(S.SelectedAnswer)).toHaveLength(0);
    question.find(S.AnswerOption).at(0).find(S.NotSelectedAnswer).simulate('click');
    expect(question.find(S.AnswerOption).at(0).find(S.NotSelectedAnswer)).toHaveLength(0);
    expect(question.find(S.AnswerOption).at(0).find(S.SelectedAnswer)).toHaveLength(1);
  });

  it('displays the correct answer to a question if given a non empty array data prop', () => {
    const question = mount(<PlayQuestion questionData={questionData} answersData={answersData} showPopup={noop} />);
    expect(question.find('div').at(5).find('p').text()).toBe('cats was correct.');
    expect(question.find('div').at(6).find('p').text()).toBe('dogs was not correct.');
  });

  // snapshots
  it('renders correctly with an empty answersData data prop', () => {
    const question = renderer.create(<PlayQuestion questionData={questionData} answersData={emptyAnswersData} showPopup={noop} />).toJSON();
    expect(question).toMatchSnapshot();
  });

  it('renders correctly with a non empty answersData data prop', () => {
    const question = renderer.create(<PlayQuestion questionData={questionData} answersData={answersData} showPopup={noop} />).toJSON();
    expect(question).toMatchSnapshot();
  });
});
