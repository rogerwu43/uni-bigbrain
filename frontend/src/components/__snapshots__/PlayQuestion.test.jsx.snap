// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`PlayQuestion renders correctly with a non empty answersData data prop 1`] = `
<div>
  <h2
    className="sc-iCfLBT sc-hBURRC ceJMkD hvAwKI"
  >
    Cats or dogs
  </h2>
  <div
    className="sc-gsDJrp ilyblg"
  />
  <div
    className="sc-fotPbf iWEGGv"
  />
  <h3
    className="sc-furvIG fYDXIl"
  >
    Select 1 Answer
  </h3>
  <div
    className="sc-gsDJrp ilyblg"
  />
  <div
    className="sc-dJjZJu hsVINA"
  >
    Time Out!
  </div>
  <div>
    <p>
      <b>
        cats
      </b>
       was correct.
    </p>
  </div>
  <div>
    <p>
      <b>
        dogs
      </b>
       was not correct.
    </p>
  </div>
</div>
`;

exports[`PlayQuestion renders correctly with an empty answersData data prop 1`] = `
<div>
  <h2
    className="sc-iCfLBT sc-hBURRC ceJMkD hvAwKI"
  >
    Cats or dogs
  </h2>
  <div
    className="sc-gsDJrp ilyblg"
  />
  <div
    className="sc-fotPbf iWEGGv"
  />
  <h3
    className="sc-furvIG fYDXIl"
  >
    Select 1 Answer
  </h3>
  <div
    className="sc-gsDJrp ilyblg"
  />
  <div
    className="sc-dJjZJu hsVINA"
  >
    Time Out!
  </div>
  <div
    className="sc-bqiQRQ bBMoAN"
  >
    <p>
      cats
    </p>
    <button
      className="sc-hKwCoD sc-crHlIS sc-fFehDp jfCzao kSklOu hAgGAw"
      onClick={[Function]}
    >
      Not Selected
    </button>
  </div>
  <div
    className="sc-bqiQRQ bBMoAN"
  >
    <p>
      dogs
    </p>
    <button
      className="sc-hKwCoD sc-crHlIS sc-fFehDp jfCzao kSklOu hAgGAw"
      onClick={[Function]}
    >
      Not Selected
    </button>
  </div>
</div>
`;
